owncloudApp/src/main/java/com/owncloud/android/presentation/ui/migration/StorageMigrationActivity.kt

```diff
@@ -0,0 +1,107 @@
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.presentation.ui.migration
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import com.owncloud.android.MainApp
+import com.owncloud.android.R
+import com.owncloud.android.data.preferences.datasources.implementation.SharedPreferencesProviderImpl
+import com.owncloud.android.data.storage.LegacyStorageProvider
+import com.owncloud.android.presentation.viewmodels.migration.MigrationState
+import com.owncloud.android.presentation.viewmodels.migration.MigrationViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import timber.log.Timber
+import java.io.File
+
+class StorageMigrationActivity : AppCompatActivity() {
+
+    private val migrationViewModel: MigrationViewModel by viewModel()
+
+    private val fragmentMigrationIntro = MigrationIntroFragment()
+    private val fragmentMigrationChoice = MigrationChoiceFragment()
+    private val fragmentMigrationProgress = MigrationProgressFragment()
+    private val fragmentMigrationCompleted = MigrationCompletedFragment()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_storage_migration)
+
+        migrationViewModel.migrationState.observe(this, {
+            navigateToNextMigrationScreen(it.peekContent())
+        })
+    }
+
+    override fun onBackPressed() {}
+
+    private fun navigateToNextMigrationScreen(migrationState: MigrationState) {
+
+        val targetFragment: Fragment = when (migrationState) {
+            is MigrationState.MigrationIntroState -> fragmentMigrationIntro
+            is MigrationState.MigrationChoiceState -> fragmentMigrationChoice
+            is MigrationState.MigrationProgressState -> fragmentMigrationProgress
+            is MigrationState.MigrationCompletedState -> fragmentMigrationCompleted
+        }
+
+        supportFragmentManager
+            .beginTransaction()
+            .replace(R.id.migration_frame_layout, targetFragment)
+            .commit()
+    }
+
+    companion object {
+
+        const val PREFERENCE_ALREADY_MIGRATED_TO_SCOPED_STORAGE = "MIGRATED_TO_SCOPED_STORAGE"
+
+        private fun hasDataInLegacyStorage(): Boolean {
+            val legacyStorageProvider = LegacyStorageProvider(MainApp.dataFolder)
+            val legacyStorageFolder = File(legacyStorageProvider.getRootFolderPath())
```

Comment:
These 2 variables are used in 2 different methods, what do you think about defining them outside of these methods as `private val`, inside the companion object?